lib/coverage_reporter/uncovered_line_grouper.ex

```diff
@@ -0,0 +1,73 @@
+defmodule CoverageReporter.UncoveredLineGrouper do
+  @moduledoc false
+
+  # Groups consecutive uncovered lines into single annotations.
+  def group_lines(source_lines, coverage_map) do
+    source_lines
+    |> Enum.map(fn {line_content, line_number} ->
+      coverage_count = Map.get(coverage_map, line_number, :no_coverage)
+      uncovered? = coverage_count == 0
+      # Only treat blank lines as bridgeable if they don't have explicit coverage data
+      blank_line? = coverage_count == :no_coverage and String.trim(line_content) == ""
+      {line_number, line_content, uncovered?, blank_line?}
```

Comment:
:information_source: Minor: given that `uncovered?` and `blank_line?` are mutually exclusive, it may make sense to use a single enum atom to cover all the possible states of a line?
```suggestion
      coverage =
        case coverage_map do
          %{^line_number => 0} ->
            :uncovered

          %{^line_number => coverage_count} when coverage_count > 0 ->
            :covered

         %{} ->
           # Only treat blank lines as bridgeable if they don't have explicit coverage data
           if String.trim(line_content) == "", do: :blank_line, else: :uncovered
      end

      {line_number, line_content, coverage}
```

Reply:
:information_source:  I think if you use Dmitry's suggestion you can skip the first case, and just do:
`%{} -> if String.trim(line_content) == "", do: :blank_line, else: :uncovered` since `when coverage_count > 0` should cover the other cases.